src/core/abstract/MCAbstractPart.cpp

```diff
@@ -260,7 +260,10 @@ void AbstractPart::importIMAPFields(struct mailimap_body_fields * fields,
                     imap_param = (struct mailimap_single_body_fld_param *) clist_content(cur);
                     
                     if (strcasecmp(imap_param->pa_name, "filename") == 0) {
-                        setFilename(String::stringByDecodingMIMEHeaderValue(imap_param->pa_value));
+                        String * filename = String::stringByDecodingMIMEHeaderValue(imap_param->pa_value);
```

Comment:
Could you check for NULL? I don't think String::stringByDecodingMIMEHeaderValue() guarantees that the result is going to be != NULL.